### PR TITLE
Some minor keyboard lock api improvements

### PR DESCRIPTION
**Pull request by jespertheend**
### Description

- Clarified how `navigator.keyboard.lock()` works when called multiple times.
- More detailed return value description.

### Motivation

The previous description for calling `lock()` multiple times was a bit unclear to me. It seems to be copied directly from [the spec](https://wicg.github.io/keyboard-lock/#h-keyboard-lock:~:text=is%20called%20multiple%20times%20without%20an%20intervening).
The 'without an intervening call to `unlock()`' part threw me off, because this made it sound like it *will* combine all previous keycodes if you were to call `unlock()` in between.

### CODE REVIEW

File: files/en-us/web/api/keyboard/lock/index.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ capture of keypresses for any or all of the keys on the physical keyboard. This
 can only capture keys that are granted access by the underlying operating
 system.
 
-If `lock()` is called multiple times without an intervening call to {{domxref("Keyboard.unlock()", "unlock()")}}, then only the key codes specified in the last call will be locked.
+If `lock()` is called multiple times, previously locked keys will be unlocked and only the key codes specified in the most recent call will be locked.
```
Comment (hamishwillee) [inline files/en-us/web/api/keyboard/lock/index.md:19]: Absolutely agree that the mention of unlock here is unhelpful. I find this a bit more readable though.

```suggestion
If `lock()` is called multiple times then only the key codes specified in the most recent call will be locked.
Any keys locked by a previous call to `lock()` are unlocked.
```

**Comment by hamishwillee** [comment]
FWIW I don't think this doc is very clear about what "lock" means. I.e.Even reading https://developer.mozilla.org/en-US/docs/Web/API/Keyboard_API#keyboard_locking I still don't know what it means. 

I think locking means that the keys will be sent to the web page even if used with modifiers, which would otherwise change the code sent. This doc should say that, and https://developer.mozilla.org/en-US/docs/Web/API/Keyboard_API#keyboard_locking should show code that captures a particular key as well as locking a key.

**Review by hamishwillee — APPROVED** [review]
Thanks @jespertheend 

merging, but I'd welcome something to address https://github.com/mdn/content/pull/30071#issuecomment-1796877517

**Comment by jespertheend** [comment]
> Even reading https://developer.mozilla.org/en-US/docs/Web/API/Keyboard_API#keyboard_locking I still don't know what it means.

Yeah honestly it's a pretty weird API. As far as I can tell it's similar to `event.preventDefault()` on keyboard events, except that this also works to prevent some system level shortcuts.
If I'm not mistaken this is only really used in the field to prevent shortcuts like 'Escape' from exiting full screen and 'Ctrl + W' from closing tabs.
I'm not entirely sure why a good ol' `preventDefault()` wasn't an option for these cases.

**Comment by jespertheend** [comment]
Ah seems like `preventDefault()` wasn't an option because of some synchronous low-level OS APIs.

http://discourse.wicg.io/t/proposal-system-keyboard-lock-api/1594/13
> and `preventDefault()` may not work on intercepted system keys due to the synchronous nature of the low-level OS APIs used to intercept them.

**Comment by hamishwillee** [comment]
Cool! Either @wbamberg or I will update these docs to better explain the API / add a live example

**Pull request merged**